src/xds/client.rs

```diff
@@ -647,10 +645,14 @@ impl AdsClient {
                     self.handle_demand_event(_demand_event, &discovery_req_tx).await?;
                 }
                 msg = response_stream.message() => {
-                    let msg = msg?;
+                    let Some(msg) = msg? else {
+                        // If we got a None message, the stream ended without error.
+                        // This could be an explicit OK response, or if the stream is reset without a gRPC status.
```

Comment:
Should we log so that we know this was a stream termination?

Reply:
It is logged above, you will see logs like:

```
2024-11-22T20:37:36.545633Z     warn    xds::client:xds{id=7}   XDS client complete
2024-11-22T20:37:36.552650Z     info    xds::client:xds{id=8}   Stream established
2024-11-22T20:37:36.553004Z     info    xds::client:xds{id=8}   received response       type_url="type.googleapis.com/istio.workload.Address" size=40 removes=0
2024-11-22T20:37:36.554415Z     info    xds::client:xds{id=8}   received response       type_url="type.googleapis.com/istio.security.Authorization" size=0 removes=0
2024-11-22T20:37:38.555672Z     warn    xds::client:xds{id=8}   XDS client complete
2024-11-22T20:37:38.568035Z     info    xds::client:xds{id=9}   Stream established
2024-11-22T20:37:38.568344Z     info    xds::client:xds{id=9}   received response       type_url="type.googleapis.com/istio.workload.Address" size=40 removes=0
2024-11-22T20:37:38.570170Z     info    xds::client:xds{id=9}   received response       type_url="type.googleapis.com/istio.security.Authorization" size=0 removes=0
```

(test has the server close it every 2s)

Reply:
Ah you mean the `XDS client complete` message? That's probably sufficient.